vault/client.go

```diff
@@ -88,3 +91,47 @@ func (c *VaultClient) WriteKey(path, key, value string) error {
 	data[key] = value
 	return c.Write(path, data)
 }
+
+// VRead reads a versioned secret from vault - cf Read, above -
+// returns the secret (map) and the version
+func (c *VaultClient) VRead(path string) (map[string]interface{}, int64, error) {
+	secret, err := c.Read(path)
+	if err != nil {
+		return nil, -1, err
+	}
+	if len(secret) == 0 {
+		// if there is no secret and no error return a empty map
+		return secret, -1, nil
+	}
+	verObj, ok := secret["metadata"].(map[string]interface{})["version"]
+	if !ok {
+		return nil, -1, errors.New("key version not found")
```

Comment:
For consistency it may be worth defining this error as a variable, or removing the `ErrorKeyNotFound`

Reply:
done